airflow/providers/google/cloud/triggers/bigquery.py

```diff
@@ -235,7 +237,11 @@ async def run(self) -> AsyncIterator[TriggerEvent]:  # type: ignore[override]
                 job_status = await hook.get_job_status(job_id=self.job_id, project_id=self.project_id)
                 if job_status["status"] == "success":
                     query_results = await hook.get_job_output(job_id=self.job_id, project_id=self.project_id)
-                    records = hook.get_records(query_results=query_results, as_dict=self.as_dict)
+                    records = hook.get_records(
```

Comment:
Do we need to await here?

Reply:
Not in the current implementation - `get_records` is a sync method

Reply:
maybe we should add a TODO here to make it async in the future

Reply:
I added a TODO comment to the the definition of `get_records` in `triggers/bigquery.py`